### PR TITLE
replace broken --recursive with a working --ignore-submodules option

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/source-to-image/issues/530

### CODE REVIEW

**Comment by bparees** [comment]
@paralin @mfojtik @soltysh ptal.

**Comment by bparees** [comment]
[test]

**Comment by paralin** [comment]
lgtm

File: cmd/s2i/main.go [hunk 1]
```diff
@@ -165,7 +165,7 @@ $ s2i build . centos/ruby-22-centos7 hello-world-app
 	cmdutil.AddCommonFlags(buildCmd, cfg)
 
 	buildCmd.Flags().BoolVar(&(cfg.RunImage), "run", false, "Run resulting image as part of invocation of this command")
-	buildCmd.Flags().BoolVar(&(cfg.DisableRecursive), "recursive", true, "Fetch all git submodules when cloning application repository")
+	buildCmd.Flags().BoolVar(&(cfg.IgnoreSubmodules), "ignore-submodules", false, "Ignore all git submodules when cloning application repository")
```
Comment (soltysh) [inline cmd/s2i/main.go]: You want to get away without keeping backwards compatibility entirely?

**Comment by soltysh** [comment]
LGTM, just one doubt.

File: cmd/s2i/main.go (hunk 1 above)
Comment (bparees) [inline cmd/s2i/main.go]: given that the flag was completely broken for anyone using it, yes :)

Comment (bparees) [inline cmd/s2i/main.go]: (I really don't want to retain deprecated backwards compatibility for something that has confusing behavior.  better to just get people to move to the right flag)

**Comment by bparees** [comment]
[merge]

**Comment by php-coder** [comment]
Looks like you forgot to update the docs: https://github.com/openshift/source-to-image/blob/master/docs/cli.md#build-flags

**Pull request merged**